### ISSUE TITLE
[Embedding] DartPad embedding using code blocks is broken

### ISSUE THREAD

**Issue by joshuadeguzman** [dart-lang/dart-pad#1555]
## What happened?

Initially, I was using gists for embedding the code samples I am creating. Unfortunately, unauthenticated requests has a rate limit of 60 / hour.

Adding authentication layer adds extra complexity for the participants who want to try out the code and resources.

So I opted to use code blocks, as described in this [wiki](https://github.com/dart-lang/dart-pad/wiki/Embedding-Guide), to overcome the rate limit issue.

Problem is there were tags generated on the embedded Dart code via code blocks:

```
</widget></widget></widget></pageitemview></statefulwidget></widget></widget></widget></widget></product></myhomepage>
```

> I've read that embedding using code blocks is fairly new, but it's best if we could track down the current issues encountered.

## Steps to reproduce problem

- Embed widget code inside Dart using code blocks using the class `language-run-dartpad:theme-dark:mode-flutter:split-75:run-true`
- Run the html website

## Additional info

Input via HTML and output Dart code:
https://gist.github.com/joshuadeguzman/aaeb682ae4f84f44c301804621ea8879

Intro talk resource:
https://joshuadeguzman.net/talks/flutter-everywhere

**Attachments**

![Screen Shot 2020-05-11 at 9 53 22 PM](https://user-images.githubusercontent.com/20706361/81569439-d806e580-93d1-11ea-88d5-8eca448d09b8.png)

### Browser

Browser: Google Chrome

Version: 81.0.4044.138 (Official Build) (64-bit)

Are you using any extensions/plugins that affect website behavior
(particularly those that affect iframes, such as ad blockers)?

No.

Are there any warnings or errors in your browser's JavaScript console?
If so, paste them below:

**Using gists**

Loading the iframes:

```
Access to XMLHttpRequest at 'https://api.github.com/gists/db904ce7cf1fa7602bb5c519ab05d8f2' from origin 'https://dartpad.dev' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

**After using code blocks**

Loading the iframes:

```
DevTools failed to load SourceMap: Could not load content for https://dartpad.dev/primer.css.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
```

Tapping `Run`:

UPDATED

Generated dart code lead from code blocks is not runnable due to the malformed/generated widget tree:

```
Error compiling to JavaScript:
Instance of 'minified:h_'
```

Old, related to #1554:
```
Access to XMLHttpRequest at 'https://dart-services.appspot.com/api/dartservices/v2/compileDDC' from origin 'https://dartpad.dev' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
embed_flutter.dart.js:1166 Error
    at Object.xg (embed_flutter.dart.js:2238)
    at k9.$1 (embed_flutter.dart.js:13010)
    at rG.fk (embed_flutter.dart.js:7082)
    at rg.$0 (embed_flutter.dart.js:6643)
    at Object.eD (embed_flutter.dart.js:1472)
    at Z.c_ (embed_flutter.dart.js:6545)
    at Object.xO (embed_flutter.dart.js:1538)
    at pE.$1 (embed_flutter.dart.js:6723)
    at qU.$1 (embed_flutter.dart.js:9759)
    at Dw (embed_flutter.dart.js:516)
embed_flutter.dart.js:9060 POST https://dart-services.appspot.com/api/dartservices/v2/compileDDC net::ERR_FAILED
```

### Machine

Operating system: MacOS Mojave

Version: 10.14.6

### Your code

https://gist.github.com/joshuadeguzman/aaeb682ae4f84f44c301804621ea8879#file-input_via_html-txt

### DartPad's output

https://gist.github.com/joshuadeguzman/aaeb682ae4f84f44c301804621ea8879#file-output-dart

**Comment by domesticmouse** [comment]
I believe this is related to https://github.com/dart-lang/dart-pad/issues/1554

**Comment by joshuadeguzman** [comment]
Hey @domesticmouse, thanks for dropping by.

The previous logs were actually related to the outage last night. I have added the new logs for the malformed generated code which could be solved easily by fixing the code, as a workaround, in the embedded DartPad itself. Although the workaround would add more complexity for some just to run the pad.

Let me know your thoughts. Thank you!

**Comment by allentv** [comment]
So I downloaded the `dart-services` repo on my machine and ran the service locally. I made a POST request via Postman to the `analyze` endpoint and can confirm that the CORS headers are available in the response.

<img width="962" alt="Screenshot 2020-05-12 at 21 49 00" src="https://user-images.githubusercontent.com/6534992/81743946-69418f00-949a-11ea-8434-21f47d9a862c.png">

The issue seems to be a problem with the server which seems to be working now.

**Comment by domesticmouse** [comment]
I seem to be having an issue with the servers going sideways after 15 odd hours of uptime. I've just added a forced-reboot after four hours to mitigate this situation. Hopefully.

**Comment by RedBrogdon** [comment]
@joshuadeguzman, do you mind having another look at your page to see if the problem persists?

I'm currently able to use each of the examples found here:

https://joshuadeguzman.net/talks/flutter-everywhere

Hopefully this was just a side effect of the server problems we had earlier this week. If not, though, we'll keep investigating.

**Comment by joshuadeguzman** [comment]
Hi @RedBrogdon, thanks for looping in. Yes, since I will be using it for the talk tonight I had use some gists temporarily. Had to use code blocks for the Dart exercises (1-3), and use gists for the Flutter (4-7) ones. Hope this helps. Thank you!

**Comment by joshuadeguzman** [comment]
I added a reproducible output:
https://joshuadeguzman.net/talks/flutter-everywhere/test-1555

Source code:
https://gist.github.com/joshuadeguzman/93ccbd7bf6f65b269b7a6da66f01a6d3

Thank you @RedBrogdon!

**Comment by joshuadeguzman** [comment]
I'll try looking into it as well over the weekend.

**Comment by domesticmouse** [comment]
If I strip the HTML wrappering the Flutter code in that gist, it works.

**Comment by domesticmouse** [comment]
Hey @joshuadeguzman, just to circle back on this, here's [your gist](https://gist.github.com/domesticmouse/2ca14a6fed8dc073625e843b16bd3c74), lightly edited. And here's [it in DartPad](https://dartpad.dev/2ca14a6fed8dc073625e843b16bd3c74).

Let me know if there is anything else I can do to help. Thanks!

**Comment by joshuadeguzman** [comment]
Thanks @domesticmouse! Yeah, it works fine really on gists + dartpad. Unfortunately, I guess it still broken using code blocks.

![Screen Shot 2020-05-28 at 9 28 17 AM](https://user-images.githubusercontent.com/20706361/83088175-99597680-a0c5-11ea-86f0-6d9eaf66b8da.png)

But yeah, I just encountered this one because I want to skip the authorisation and rate limit issues.

Should we still keep this issue open though as for the code blocks issue reference in the future?

**Comment by domesticmouse** [comment]
Ahh, my bad. I missed that you were trying to get `inject_embed.dart.js` to work properly. I'll assign this to @johnpryan who knows way more about this tech than I do. Thanks!

**Comment by joshuadeguzman** [comment]
No worries @domesticmouse, sorry if I wasn't clear about embedding (injecting) code blocks. But hey, thanks for helping out!

**Comment by parlough** [comment]
Thanks for all of the discussion on this issue. 

The newest DartPad frontend supports embed injection in a different way, and while it's not officially supported, https://github.com/dart-lang/site-shared/tree/main/packages/inject_dartpad has some details on how to use it.

If you have any issues or questions with embeds using the new frontend, please open a new issue and I'd be happy to help. Thanks!
